experimental/InjectiveResolutions/src/LocalCohomology.jl

```diff
@@ -0,0 +1,642 @@
+        push!(columns, _psi_del[:, i - k])
+      end
+      if length(columns) > 0
+        psi_del = matrix(field, hcat(columns...))
+      end
+    end
+  end
+
+  #define maps phi_{0,1} and psi_{1,2}
+  f0 = hom(F_0, F_1, phi_del)
+  f1 = hom(F_1, F_2, psi_del)
+  return quo(kernel(f1)[1], image(f0)[1])[1], _A
+end
+
+@doc raw"""
+  apply_gamma!(J0::InjMod,J1::InjMod,J2::InjMod,phi::MatElem{T},psi::MatElem{T},I::MonoidAlgebraIdeal) where {T<:FieldElem}
+
+Apply the functor $\Gamma_I(-)$ to a cochain complex of injective modules $J_0 \xrightarrow{phi} J_1 \xrightarrow{\psi} J_2$. It maps a $\mathbb{Z}^d$-graded $k[Q]$-momdule $M$
+to the submodule $\Gamma_I(M) = \{m \in M \mid m \cdot I^n = 0 \text{ for some }n>0\}$. Applying the functor corresponds to deleting all indecomposable injectives
+$k\{a_i + F_i - Q\}$ with $I \not\subseteq p_{F_i}$. The monomial matrices of the maps $\phi$ and $\psi$ are updated by deleting the corresponding rows and columns.  
+"""
+function apply_gamma!(
+  J0::InjMod,
+  J1::InjMod,
+  J2::InjMod,
+  phi::MatElem{T},
+  psi::MatElem{T},
+  I::MonoidAlgebraIdeal,
+) where {T<:FieldElem}
+  @assert J0.monoid_algebra == J1.monoid_algebra == J2.monoid_algebra == I.monoid_algebra
+
+  k = coefficient_ring(I.monoid_algebra)
+  J_0 = J0.indec_injectives
+  J_1 = J1.indec_injectives
+  J_2 = J2.indec_injectives
+
+  _J_0 = []
+  rows_phi = []
+  for i in eachindex(J_0)
+    if issubset(underlying_ideal(I), J_0[i].face.prime)
+      # if issubset(I,J_0[i].face.prime) 
+      push!(_J_0, J_0[i])
+      push!(rows_phi, phi[i, :])
+    end
+  end
+  _phi = transpose(hcat(rows_phi...))
+
+  columns_phi = []
+  rows_psi = []
+  _J_1 = []
+  for i in eachindex(J_1)
+    if issubset(underlying_ideal(I), J_1[i].face.prime)
+      push!(_J_1, J_1[i])
+      if rows_phi != []
+        push!(columns_phi, _phi[:, i])
+      end
+      push!(rows_psi, psi[i, :])
+    end
+  end
+  if is_empty(columns_phi)
+    phi = zero_matrix(k, length(rows_phi), 0)
+  else
+    phi = matrix(k, hcat(columns_phi...))
+  end
+  _psi = transpose(hcat(rows_psi...))
+
+  columns_psi = []
+  _J_2 = []
+  for i in eachindex(J_2)
+    if issubset(underlying_ideal(I), J_2[i].face.prime)
+      push!(_J_2, J_2[i])
+      if rows_psi != []
+        push!(columns_psi, _psi[:, i])
+      end
+    end
+  end
+  if is_empty(columns_psi)
+    psi = zero_matrix(k, length(rows_psi), 0)
+  else
+    psi = matrix(k, hcat(columns_psi...))
+  end
+
+  return [_J_0..., _J_1..., _J_2...], phi::MatElem, psi::MatElem, (length(_J_0), length(_J_0) + length(_J_1))
+end
+
+@doc raw"""
+  maps_needed(kQ::MonoidAlgebra, S_A::Vector{SectorLC})
+
+This follows Algorithm 6.4. in [HM05](@cite) and computes all needed maps (as described in the proof of [HM05](@cite) (Proposition 5.1)).
+"""
+function maps_needed(kQ::MonoidAlgebra, S_A::Vector{SectorLC})
+  k = coefficient_ring(kQ)
+  maps = []
+  for s1 in S_A, s2 in S_A # loop over all pairs
+    if s1 == s2
+      continue
+    end
```

Comment:
This is fine, but more idiomatic would be this:
```suggestion
    s1 == s2 && continue
```